workflow.txt

```diff
@@ -1,10 +1,8 @@
+fork on github
 git clone
+git pull upstream master
 git checkout -b my-solution
-git commit -m ""
-git fetch --all
 git add .
-git pull upstream master
-git merge my-solution
-fork on github
+git commit -m ""
 git checkout master
```

Comment:
Close, you would get an error trying to push to a different branch from master... in this scenario we are looking for something like: 
 - fork on github
 - git clone
 - git checkout -b my-solution
 - git add .
 - git commit -m ""
 - git push origin my-solution

that would be the flow to create a general pull request, i.e. HW submission.